### PR TITLE
Clean up types and expose more types to users

### PR DESCRIPTION
**Pull request by drwpow**
## Changes

1. Combines `astro-core` and `astro-runtime`. Originally separated because we thought we’d have a lot of types, but after the cleanup it’s not much at all. Combined into one `astro` file.
2. Deletes duplicate/unused types (except those that seemed to be exported for user benefit)
3. Adds JSDoc comments to the public-facing types
4. Renames public-facing types to be easier to find (e.g. `Page` or `RSS`)

## Testing

Tests should pass

## Docs

Docs added in the form of JSDoc for types

### CODE REVIEW

File: packages/astro/src/@types/astro.ts [hunk 1]
```diff
@@ -13,8 +21,35 @@ export interface AstroComponentMetadata {
 }
 
 /**
- * The Astro User Config Format:
- * This is the type interface for your astro.config.mjs default export.
+ * Astro.* available in all components
+ * Docs: https://docs.astro.build/reference/api-reference/#astro-global
```
Comment (drwpow) [inline packages/astro/src/@types/astro.ts:25]: Screen Shot 2021-11-12 at 17 40 16

Adding a URL makes finding the API docs easier

File: packages/astro/src/@types/astro.ts [hunk 2]
```diff
@@ -167,33 +198,24 @@ export interface ManifestData {
...
+  /** object of props to forward to `page` result */
+  props?: Props;
 }
 
-export interface PaginatedCollectionResult<T = any> {
+/**
+ * Page Prop
+ * Docs: https://docs.astro.build/guides/pagination/#using-the-page-prop
+ */
+export interface Page<T = any> {
```
Comment (drwpow) [inline packages/astro/src/@types/astro.ts:218]: One example of renaming `PaginatedCollectionProps` to simply `Page` for users to find more easily.

**Comment by matthewp** [comment]
If you rebase with `main` this should pass again now.

**Review by matthewp — APPROVED** [review]
lgtm

**Pull request merged**